contracts/ZeroOne/UserGroups/UserGroups.sol

```diff
@@ -0,0 +1,121 @@
+pragma solidity 0.6.1;
+pragma experimental ABIEncoderV2;
+
+import "./lib/UserGroupsList.sol";
+import "./lib/UserGroup.sol";
+import "../../Interfaces/IERC20.sol";
+
+/**
+ * @title UserGroups
+ * @dev stores user groups
+ */
+contract UserGroups {
+  using UserGroupsList for UserGroupsList.List;
+  using UserGroup for UserGroup.Group;
+
+  UserGroupsList.List groups;
+
+  event UserGroupAdded(
+    uint id,
+    string name,
+    address groupAddress
+  );
+
+  /**
+   * @notice reverts on non-existing group id
+   * @param _id group id
+  */
+  modifier groupExists(
+    uint _id
+  ) {
+    require(
+      groups.checkId(_id),
+      "Provided index is out of bounds"
+    );
+    _;
+  }
+
+  constructor() public {}
+
+  /**
+   * @notice get group data
+   * @param _id group id
+   * @return group
+  */
+  function getUserGroup(
+    uint _id
+  )
+    public
+    view
+    groupExists(_id)
+    returns (UserGroup.Group memory group)
+  {
+    return groups.list[_id];
+  }
+
+  /**
+    @notice get user group token totalSupply
+    @param _groupId group id
+    @return totalSupply
+   */
+  function getTotalSupply(uint _groupId)
+    public
+    view
+    groupExists(_groupId)
+    returns(uint256 totalSupply) 
+  {
+    UserGroup.Group memory userGroup = getUserGroup(_groupId);
```

Comment:
check gas use for group[_groupId]

Reply:
move it to UserGroup as getTotalSupply() or totalSupply()

Reply:
moved, but gas not checked

Reply:
> check gas use for group[_groupId]

i think we don't need to do this?